config/routes.rb

```diff
@@ -3,4 +3,8 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show'
+  get '/users/:user_id/posts', to: 'posts#index'
+  get '/users/:user_id/posts/:id', to: 'posts#show'
```

Comment:
- [ ] [OPTIONAL] You can use the following code snippet to represent exactly what you wrote instead of hard coding the routes.
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```